packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,416 @@
+                    self.httpd.serve_forever()
+                except OSError:
+                    logger.error(
+                        f"{self.host}:{self.port} is already in use, please try another Socket."
+                    )
+
+    def disconnect(self) -> None:
+        """
+        Disconnect.
+
+        Shut-off the HTTP Server and then HTTP Channel.
+        """
+        with self.lock:
+            if not self.stopped:
+                self.httpd.shutdown()
+                logger.info(f"HTTP Server has shutdown on port {self.port}.")
+                self.stopped = True
+                # self.thread.join()
+
+
+def HTTPHandlerFactory(channel: HTTPChannel):
+    class HTTPHandler(BaseHTTPRequestHandler):
+        def __init__(self, *args, **kwargs):
+            self._channel = channel
+            super(HTTPHandler, self).__init__(*args, **kwargs)
+
+        def do_HEAD(self):
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+
+        def do_GET(self):
+            """Respond to a GET request."""
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+
+            parsed_path = urlparse(self.path)
+            url = parsed_path.path
+            # url = parsed_path.geturl()
+            param = parse_qs(parsed_path.query)
+            param = json.dumps(param)
+
+            self._send("GET", url, param)
+
+        def do_POST(self):
+            """Respond to a POST request."""
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+
+            parsed_path = urlparse(self.path)
+            url = parsed_path.path
+            # url = parsed_path.geturl()
+            param = parse_qs(parsed_path.query)
+            param = json.dumps(param)
+
+            content_length = int(self.headers["Content-Length"])
+            body = self.rfile.read(content_length).decode()
+
+            self._send("POST", url, param, body)
+
+        def _send(
+            self, http_method: str, url: str, param: str, body: Optional[str] = r"{}"
+        ):
+            api_req_envelope = self._channel._process_incoming_request(
+                http_method, url, param, body
+            )
+            response = self._channel._helper._verify(api_req_envelope)
+            # Write the API response back to console
+            console_response = f"\n{response}\n\n"
+            self.wfile.write(console_response.encode())
+            # Send the Envelope to the Agent's InBox.
+            # assert self._channel.in_queue is not None, "Input queue not initialized."
+            # assert self._channel.loop is not None, "Loop not initialized."
+            # self._channel.send(api_req_envelope)
+
+    return HTTPHandler
+
+
+class APIVerificationHelper:
+    def __init__(self, e: Optional[Envelope]):
```

Comment:
this should consume the api_specification_path, not the envelopes

Reply:
fixed, also removed function _process_initial_api() as it is not needed anymore.